AIM/AIM/UI/Views/TrackCell.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  TrackCell.swift
+//  AIM
+//
+//  Created by Oskar Szydlowski on 28.02.2018.
+//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
+//
+
+import UIKit
+import Kingfisher
+
+class TrackCell: UITableViewCell {
+    
+    private let titleLabel: UILabel = {
+        let label = UILabel()
+        label.textColor = UIColor.brown
+        return label
+    }()
+    private let artistLabel = UILabel()
+    private let coverImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.backgroundColor = UIColor.lightGray
+        return imageView
+    }()
+    private let durationLabel = UILabel()
+    
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        selectionStyle = .none
+        setupView()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
```

Comment:
There is no preapreForReuse implemented. It should cancel image fetching task request. Kingfisher returns `RetrieveImageTask` which you should store in cell and use it to cancel fetching. Whole extension for imageview can look like this:

```swift
import UIKit
import Kingfisher

protocol ImageTask {
    func cancel()
}

extension RetrieveImageTask: ImageTask { }

extension UIImageView {

    @discardableResult func setImage(with url: URL?) -> ImageTask {
        return kf.setImage(with: url)
    }
}

```